internal/common/connectionsapi/client.go

```diff
@@ -0,0 +1,147 @@
+package connectionsapi
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+type Client struct {
+	authToken string
+	apiURL    url.URL
+	client    *http.Client
+	userAgent string
+}
+
+const (
+	defaultRetries = 3
+	defaultTimeout = 90 * time.Second
+	pathPrefix     = "/api/v1/stacks"
+)
+
+func NewClient(authToken string, rawURL string, client *http.Client, userAgent string) (*Client, error) {
+	parsedURL, err := url.Parse(rawURL)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse connections API url: %w", err)
+	}
+
+	if client == nil {
+		retryClient := retryablehttp.NewClient()
+		retryClient.RetryMax = defaultRetries
+		client = retryClient.StandardClient()
+		client.Timeout = defaultTimeout
+	}
+
+	return &Client{
+		authToken: authToken,
+		apiURL:    *parsedURL,
+		client:    client,
+		userAgent: userAgent,
+	}, nil
+}
+
+type apiResponseWrapper[T any] struct {
+	Data T `json:"data"`
+}
+
+type MetricsEndpointScrapeJob struct {
+	Enabled                     bool   `json:"enabled"`
+	AuthenticationMethod        string `json:"authentication_method"`
+	AuthenticationBearerToken   string `json:"bearer_token,omitempty"`
+	AuthenticationBasicUsername string `json:"basic_username,omitempty"`
+	AuthenticationBasicPassword string `json:"basic_password,omitempty"`
+	URL                         string `json:"url"`
+	ScrapeIntervalSeconds       int64  `json:"scrape_interval_seconds"`
+}
+
+func (c *Client) CreateMetricsEndpointScrapeJob(ctx context.Context, stackID, jobName string, jobData MetricsEndpointScrapeJob) (MetricsEndpointScrapeJob, error) {
```

Comment:
see https://github.com/grafana/terraform-provider-grafana/pull/1869 for similar changes, we can remove the pointer from all of these receiver functions